### PR TITLE
[F] Document SearchResult API endpoint

### PR DESCRIPTION
**Pull request by ConnorChristensen**

### CODE REVIEW

**Comment by ConnorChristensen** [comment]
I'm getting a failure on this test that re-indexing is not working. It looks like it is looking for a connection to an elastic search instance that is not working. I'm not sure how to proceed with this problem in a timely manner. These tests work locally.

I see that the tests for the [search serializer](https://github.com/ManifoldScholar/manifold/blob/master/api/spec/serializers/search_result_serializer_spec.rb#L10) re-indexes for the tests, and those seem to work fine on the github actions test. I originally thought that there might not be an instance of elastic search running, but it seems like the search serializer test shows that it can work. I tried to copy the search serializer as best I could, but there are likely things going on behind the scenes that a more experienced ruby person could figure out faster.

If you have any hints for me, I'm happy to make another attempt at solving the failing test.

**Comment by zdavis** [comment]
We don't run tests that require a connection to ES on CI — we haven't gotten around to getting ES spun up in CI env. Tag the test with :elasticsearch and it won't run on CI. You'll see this in other places in the specs. For example:

```
  context "can be searched", :elasticsearch do
  end
```

File: api/spec/requests/api/v1/search_results_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require "swagger_helper"
+
+RSpec.describe "Search Results", type: :request do
+
+  context "can search a project", :elasticsearch do
```
Comment (ConnorChristensen) [inline api/spec/requests/api/v1/search_results_spec.rb]: Did I do this right? It looks like this test is still trying to run and failing.

**Comment by zdavis** [comment]
Rebased on Master.

**Comment by zdavis** [comment]
Changed base to master.

File: api/spec/requests/api/v1/search_results_spec.rb (hunk 1 above)
Comment (zdavis) [reply api/spec/requests/api/v1/search_results_spec.rb:5]: The tag was correct. The issue was with your spec — you were reindexing models outside of a before block and that code was being executed whether or not the examples were included. Generally, doing things outside of let and before/after blocks in specs is an antipattern and you shouldn't do it.

Comment (zdavis) [reply api/spec/requests/api/v1/search_results_spec.rb:4]: See line 9 below for the correct way to do what you were trying to do.

**Pull request merged**